### ISSUE TITLE
Extra hover text not work in Candlestick charts

### ISSUE THREAD

**Issue by Yevgnen** [plotly/plotly.js#2481]
Example:

```python
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import numpy as np
import pandas as pd
import plotly.graph_objs as go
import plotly.offline as py


import datetime

n = 20
open = 2 + np.abs(np.random.randn(n))
high = open + 1
low = open - 1
close = open + np.random.randn(n)
df = pd.DataFrame(
    {
        'open': open,
        'high': high,
        'low': low,
        'close': close
    },
    index=pd.date_range(
        '2010-01-01',
        pd.to_datetime('2010-01-01') + datetime.timedelta(days=n - 1)))

hovertext = []
for i in range(len(df.open)):
    hovertext.append('Point' + str(i))

trace = go.Candlestick(
    x=df.index,
    open=df.open,
    high=df.high,
    low=df.low,
    close=df.close,
    text=hovertext,
    hoverinfo='all')
data = [trace]
py.plot(data, filename='ohlc_custom_hover.html')
```

However, turning the line `trace = go.Candlestick(` to `trace = go.Ohlc(` works just fine.

The plotly version is `2.2.3`.

**Comment by etpinard** [comment]
Hmm. Here's another very annoying finance chart bug. Candlestick traces generate `box` traces. Box traces have a `text` attribute, but those text values correspond to the box' sample values and are shown only when only over box points. In other word, text value do no appear in hover labels under boxes. So there's no way to just push `text` info the generate box traces.

It feels like I've been saying this a million times, but really, we should get rid of our transform trace hacks.

**Comment by etpinard** [comment]
Done in https://github.com/plotly/plotly.js/pull/2561

**Comment by alexcjohnson** [comment]
Fixed by #2510 - this is working now on master.
